src/Utilities/CssConcatenator.php

```diff
@@ -150,7 +159,12 @@ private static function getMediaRuleCss(\stdClass $mediaRule): string
      */
     private static function getRuleBlockCss(\stdClass $ruleBlock): string
     {
-        $selectors = \array_keys($ruleBlock->selectorsAsKeys);
-        return \implode(',', $selectors) . '{' . $ruleBlock->declarationsBlock . '}';
+        /** @var array<string, array-key> $selectorsAsKeys */
+        $selectorsAsKeys = $ruleBlock->selectorsAsKeys;
+        /** @var array<array-key, string> $selectors */
+        $selectors = \array_keys($selectorsAsKeys);
```

Comment:
With the type hint now added for `$selectorsAsKeys`, is the one for `$selectors` necessary any more - is Psalm not able to determine it from the signature or `array_keys`, or is it unable to determine that `array_keys` will necessarily return an array of strings when the input array has string keys?

Reply:
Usually, Psalm complains if deems a type annotation to be redundant. In this case, it has not complained. :-)

Reply:
OK, I guess then it's not smart enough to deduce that `array_keys` will return an array whose values will have the same type as the keys in the input array.